iothub/service/src/HttpTransportSettings.cs

```diff
@@ -27,5 +31,20 @@ public HttpTransportSettings()
         /// especially when MQTT and AMQP ports are disallowed to the internet.
         /// </remarks>
         public IWebProxy Proxy { get; set; }
+
+        /// <summary>
+        /// An option that specifies the number of milliseconds that an active ServicePoint connection remains open for the singleton HttpClient
+        /// instances used by this client to make all HTTPS requests. The default is -1, which allows an active ServicePoint connection to stay 
+        /// connected indefinitely. Set this property to 0 to force ServicePoint connections to close after servicing a request.
+        /// For additional details, <see href="https://docs.microsoft.com/en-us/dotnet/api/system.net.servicepoint.connectionleasetimeout?view=net-5.0#System_Net_ServicePoint_ConnectionLeaseTimeout"> see here</see> 
+        /// </summary>
+        /// <remarks>
+        /// Generally it isn't advised to set this value to either -1 or 0. In the case where this value is set to -1, the singleton HTTP client will indefinitely
+        /// cache the DNS of the host. If the host has any sort of failover, the singleton HTTP client will not be made aware of the DNS change and will 
+        /// continue to send the requests to the now disabled host rather than the backup host. In the case where this value is set to 0, each HTTPS request will close it's socket when finished, rather than allowing for the re-use of sockets for
+        /// subsequent requests. This lack of caching will add a small amount of time to each HTTPS request. For applications that want both the benefits of socket caching and resiliency to DNS changes, this value is advised to be greater than 0. Higher values
+        /// will provide more socket caching performance benefits, while lower values will respond more quickly to DNS changes.
+        /// </remarks>
+        public int ConnectionLeaseTimeoutMilliseconds { get; set; } = DefaultConnectionLeaseTimeoutMilliseconds;
```

Comment:
The type of this option matches the type that we set within our SDK. I could see an argument to be made for making this a timespan, but considering how 0 and -1 are the special cases, it would feel weird to ask users construct a timespan. I'm open to suggestions though